services/templates/service.go

```diff
@@ -38,17 +36,34 @@ type Service interface {
 	ParseTemplate(c context.Context, templateID int64, userID int64) (*entities.CampaignTemplateData, error)
 }
 
+type Opts func(s *service)
+
+// SetTemplateBucket this is optionally adding template bucket for testing or for the fixture cli
+func SetTemplateBucket(bucket string) Opts {
```

Comment:
Since we're going with "functional options" it's better to name them to something like:

```suggestion
func TemplateBucket(bucket string) Opts {
```

And the init code will look like this:

```go
New(db, s3, svc.TemplateName("foobar"))
```